src/sources/http_scrape/scrape.rs

```diff
@@ -0,0 +1,241 @@
+//! Generalized HTTP scrape source.
+//! Scrapes an endpoint at an interval, decoding the HTTP responses into events.
+
+use bytes::{Bytes, BytesMut};
+use chrono::Utc;
+use futures_util::FutureExt;
+use http::{response::Parts, Uri};
+use snafu::ResultExt;
+use std::collections::HashMap;
+use tokio_util::codec::Decoder as _;
+
+use crate::{
+    codecs::{Decoder, DecodingConfig},
+    config::{SourceConfig, SourceContext, SourceDescription},
+    http::Auth,
+    serde::default_decoding,
+    serde::default_framing_message_based,
+    sources,
+    tls::{TlsConfig, TlsSettings},
+    Result,
+};
+use codecs::{
+    decoding::{DeserializerConfig, FramingConfig},
+    StreamDecodingError,
+};
+use vector_config::configurable_component;
+use vector_core::{
+    config::{log_schema, LogNamespace, Output},
+    event::Event,
+};
+
+/// The name of this source
+pub(crate) const NAME: &str = "http_scrape";
+
+/// Configuration for the `http_scrape` source.
+#[configurable_component(source)]
+#[derive(Clone, Debug)]
+pub struct HttpScrapeConfig {
+    /// Endpoint to scrape events from.
+    endpoint: String,
```

Comment:
It's unclear immediately if this includes a path, can we clarify?

Reply:
Hmm my understanding of an endpoint = Base URL + Resources + Params

so `http://foo.com/the/resource/and?params=one+two` , would be an endpoint in my book.

In that construction is a Resource (the/resource) , the same as your Path ?

Reply:
Yeah - I think just pulling the "full" description + warning from cue here is good

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/13793/commits/5f7c44231f801eb5ee0be2f2a3549efb6c908e8c